qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,619 @@
+    implementation as it will likely include additional optimizations and be a
+    more efficient implementation. The generic nature of this class precludes
+    doing any provider- or backend-specific optimizations.
+    """
+
+    # pylint: disable=missing-raises-doc
+    def __init__(
+        self,
+        backend: Backend,
+        *,  # TODO: allow backend as positional after removing deprecations
+        abelian_grouping: bool = True,
+        skip_transpilation: bool = False,
+        bound_pass_manager: PassManager | None = None,
+        options: dict | None = None,
+    ):
+        """Initalize a new BackendEstimator instance.
+
+        Args:
+            backend: Required: The backend to run the primitive on.
+            abelian_grouping: Whether commuting observable components should be grouped.
+            skip_transpilation: If `True`, transpilation of the input circuits is skipped.
+            bound_pass_manager: An optional pass manager to run after parameter binding.
+            options: Default options.
+        """
+        # TODO: validation
+        self.backend = backend
+        self.abelian_grouping = abelian_grouping
+        self.skip_transpilation = skip_transpilation  # TODO: tranpilation level
+        self._bound_pass_manager = bound_pass_manager  # TODO: standardize
+        super().__init__(
+            circuits=None,
+            observables=None,
+            parameters=None,
+            options=options,
+        )
+        self._transpile_options = Options()
+
+    ################################################################################
+    ## PROPERTIES
+    ################################################################################
+    @property
+    def backend(self) -> Backend:  # TODO: normalize to one type of Backend
+        """Backend to use for circuit measurements."""
+        return self._backend
+
+    @backend.setter
+    def backend(self, backend: Backend) -> None:
+        if not isinstance(backend, Backend):
+            raise TypeError(
+                "Expected `Backend` type for `backend`, " f"got `{type(backend)}` instead."
+            )
+        # TODO: clear all transpilation caching
+        if isinstance(backend, BackendV1):
+            backend.max_circuits = getattr(backend.configuration(), "max_experiments", None)
+        self._backend = backend
+
+    @property
+    def abelian_grouping(self) -> bool:
+        """Groups commuting observable components."""
+        try:
+            return self._abelian_grouping
+        except AttributeError:
+            return True
+
+    @abelian_grouping.setter
+    def abelian_grouping(self, abelian_grouping: bool) -> None:
+        if not isinstance(abelian_grouping, bool):
+            raise TypeError(
+                "Expected `bool` type for `abelian_grouping`, "
+                f"got `{type(abelian_grouping)}` instead."
+            )
+        self._abelian_grouping = abelian_grouping
```

Comment:
What's wrong with just using the base property, and why do you need the `try/except`?  You've type-hinted it, so it's already clear what values are incorrect, and the constructor unconditionally sets the attribute.

Reply:
I use the property for validation, trying to provide users with an early understanding of why things fail. 

It is really a matter of personal preference, I'd rather inform as soon as possible (especially if the check is not particularly costly) so that users can learn the tool by interacting with it as opposed to strictly needing the docs to avoid unexpected side-effects. This is also useful when integrating calls like this in another function.

As for the `try/except` you are right, it is not strictly needed but I like to put it there as a safeguard in case developers choose to bypass the initializer for some reason.

Reply:
I removed it for now, we'll see later if we need it.